PasswordVerifier/src/main.java

```diff
@@ -0,0 +1,134 @@
+                errorIndicies.add(new Integer(i));
+        }
+        if(errorIndicies.size() == 0) {
+            System.out.println("no errors. Good job!");
+            return true;
+        }
+        else {
+            System.out.println("errors at indicies:");
+            System.out.println(errorIndicies.toString());
+            return false;
+        }
+    }
+
+    /*
+     * stress tests isValid function. prints time to check test cases
+     * @param count number of test cases to run
+     */
+    public static void stressTest(int count, boolean logging) {
+        final int minLength = 5;
+        final int maxLength = 20;
+        int nextLength;
+        char nextChar;
+        char[] nextString;
+        Random myRandom = new Random();
+        String[] testCases = new String[count];
+        boolean[] testResults = new boolean[count];
+        for (int i = 0; i < testCases.length; ++i) {
+            nextLength = (myRandom.nextInt(maxLength - minLength + 1) + minLength);
+            nextString = new char[nextLength];
+            for (int j = 0; j < nextString.length; ++j) {
+                //arbitrary ascii range containing mostly valid characters
+                nextChar = (char) (myRandom.nextInt(100) + 27);
+                nextString[j] = nextChar;
+            }
+            testCases[i] = new String(nextString);
+        }
+        
+        long startTime = System.nanoTime();
+        for (int i = 0; i < testCases.length; ++i) {
+            testResults[i] = utils.isValid(testCases[i]);
+        }
+        long endTime = System.nanoTime();
+        long duration = (endTime - startTime);
+        System.out.println("time to check: " + duration + " nanoseconds");
+        
+        if(logging) {
+	        PrintWriter myWriter = null;
+	        try {
+	        	myWriter = new PrintWriter("log.txt", "UTF-8");
+	        	for (int i = 0; i < testCases.length; ++i) {
+	        		myWriter.println(testCases[i] + ": " + testResults[i]);
+	        	}
+	        } catch (FileNotFoundException e) {
+	        	e.printStackTrace();
+	        } catch (UnsupportedEncodingException e) {
+	        	e.printStackTrace();
+	        } finally {
+	        	myWriter.close();
+	        }
+        }
+    }
+    
+    public static void main(String[] args) {
+    	Properties myProperties = new Properties();
+    	FileInputStream input = null;
+    	boolean logging = false;
+    	try {
+    		input = new FileInputStream("/Users/benjaminkuschner/Documents/SecurityProjects/PasswordVerifier/src/config.properties");
+    		myProperties.load(input);
+    		if(myProperties.getProperty("logging").equals("true"))
+				logging = true;
+    		else
+    			logging = false;
+    	} catch (IOException ex) {
+    		ex.printStackTrace();
+    	} finally {
+    		if (input != null) {
+    			try {
+    				input.close();
+    			} catch(IOException e ) {
+    				e.printStackTrace();
+    			}
+    		}
+    	}
+    	
+        correctnessTest();
```

Comment:
correctnessTest returns a boolean, that indicates we want to do something different if it passes or fails - perhaps stop or at least log an error? Perhaps there's logging inside correctnessTest, but one master error indicating that it failed could be useful for troubleshooting